File: appfw/src/main/java/com/jx/appfw/domain/interactor/main/TestAction.java
package com.jx.appfw.domain.interactor.main;

import com.jx.appfw.domain.repository.MainRepository;
import com.jx.appfw.domain.request.main.TestBean;
import com.jx.arch.domain.executor.PostExecutionThread;
import com.jx.arch.domain.executor.ThreadExecutor;
import com.jx.arch.domain.interactor.UseCase;

import javax.inject.Inject;

import io.reactivex.Observable;

public class TestAction extends UseCase<String, TestBean> {

    private final MainRepository repository;

    @Inject
    TestAction(MainRepository repository,
               ThreadExecutor threadExecutor, PostExecutionThread postExecutionThread) {
        super(threadExecutor, postExecutionThread);
        this.repository = repository;
    }

    @Override
    public Observable<String> buildUseCaseObservable(TestBean testBean) {
        return repository.test(testBean);

    }
}

File: appfw/src/main/java/com/jx/appfw/domain/interactor/main/TestAction.kt
package com.jx.appfw.domain.interactor.main;

import com.jx.appfw.common.JobExecutor
import com.jx.appfw.common.PostExecutionThread
import com.jx.appfw.common.UseCase
import com.jx.appfw.domain.repository.MainRepository
import com.jx.appfw.domain.request.main.TestBean
import io.reactivex.Observable

class TestAction(
    val mMainRepository: MainRepository,
    mThreadExecutor: JobExecutor,
    mPostExecutionThread: PostExecutionThread
) : UseCase<String, TestBean>(mThreadExecutor, mPostExecutionThread) {

    @Override
    override fun buildUseCaseObservable(testBean: TestBean): Observable<String> {
        return mMainRepository.test(testBean)
    }
}
